subcommands/keys/ca_revoke_device_ca.go

```diff
@@ -113,6 +114,31 @@ func doRevokeDeviceCa(cmd *cobra.Command, args []string) {
 		ca := x509.LoadCertFromFile(filename)
 		toRevoke[ca.SerialNumber.Text(10)] = crlReason
 	}
+
+	caList, err := api.FactoryGetCA(factory)
+	subcommands.DieNotNil(err)
+	validSerials := make(map[string]bool, 0)
+	for len(caList.CaCrt) > 0 {
+		block, remaining := pem.Decode([]byte(caList.CaCrt))
+		if block == nil {
+			// could be excessive whitespace
+			caList.CaCrt = strings.TrimSpace(string(remaining))
```

Comment:
It looks like an infinite loop may occur if the `remaining` is some garbage so `block` is `nil` while `remaining` has no spaces just some garbage.

Reply:
Hmm... I think you are right: _If no PEM data is found, p is nil and the whole of the input is returned in rest._
Which means that the original code in `show.go` where I copied it from has the same problem;.

So, I need to fix both paces now. Good catch!

Reply:
A fix https://github.com/foundriesio/fioctl/pull/324/commits/94ca3f22dfc8a98792a928bac9efe8bae23d2098, a cleanup https://github.com/foundriesio/fioctl/pull/324/commits/065dbb7f5129cc05fd2d3de389fe24ca0fd2289e, and a fixup (to be squashed) https://github.com/foundriesio/fioctl/pull/324/commits/762be5a68614b6186e001fd595dced9c13a7e285.